app.js

```diff
@@ -35,41 +34,68 @@ addMushroomButton.addEventListener('click', () => {
 
         mushroomCount++;
         displayMushrooms();
+
     } else {
         alert('no luck!');
     }
+    displayMushrooms();
+
 });
 
 addFriendButton.addEventListener('click', () => {
     // get the name from the input
-    // create a new friend object
-    // push it into the friends state array, passed in as an argument
-    // reset the input
-    // display all the friends (use a function here)
+    let newFriendName = friendInputEl.value || 'Friendo';
+
+    let newFriend = {
+        name: newFriendName,
+        satisfaction: 1
+    };
+
+    friendData.push(newFriend);
+
+    displayFriends();
+    friendInputEl.textContent = '';
 });
 
 function displayFriends() {
-    // clear out the friends in DOM
 
-    // for each friend in state . . .
+    friendsEl.textContent = '';
+
     for (let friend of friendData) {
-        // use renderFriend to make a friendEl
+        const aFriend = renderFriend(friend);
+
+        aFriend.addEventListener('click', () => {
+            if (friend.satisfaction < 3 && mushroomCount > 0) {
+                friend.satisfaction++;
+                mushroomCount--;
+            }
+            
+            else if (mushroomCount === 0) {
+                alert('There aren\'t any more mushrooms to share! Time to forage for more!');
+            }
 
-        // this is a clickable list, so . . .
-        //     add an event listener to each friend
-        //         and if the friend's satisfaction level is below 3 and you have mushrooms left
-        //             increment the friends satisfaction and decrement your mushrooms
-        //             then display your friends and mushrooms with the updated state
+            else if (friend.satisfaction === 3) {
+                alert(`${friend.name}'s had plenty of mushrooms already! Try sharing with someone else, or invite a new friend!`);
+            }
+            
+            displayFriends();
+            displayMushrooms();
+        });
 
-        // append the friendEl to the friends list in DOM
+        friendsEl.append(aFriend);
     }
 }
 
+
 function displayMushrooms() {
-    // clear out the mushroom div
+
+    mushroomsEl.textContent = '';
 
     for (let i = 0; i < mushroomCount; i++) {
-        // for each mushroom in your mushroom state, render and append a mushroom
+        const mushroom = document.createElement('div');
+        mushroom.classList.add('mushroom');
+        renderMushroom(mushroom);
+        mushroomsEl.append(mushroom);
     }
```

Comment:
Line 97 here made me take a closer look at the rest of this. It looks like you're calling the function`renderMushroom(mushroom)` which means it should be returning a value (a DOM element in this case), but you're not doing anything with that value that is returned. 

I believe all that you should need here would be the following:

```js
for (let i = 0; i < mushroomCount; i++) {
        const mushroom = renderMushroom(mushroom);
        mushroomsEl.append(mushroom);
    }
```